pkg/vulnerability/vulnerability.go

```diff
@@ -120,7 +131,45 @@ func (c Client) Filter(vulns []types.DetectedVulnerability, severities []dbTypes
 		}
 		return vulnerabilities[i].VulnerabilityID < vulnerabilities[j].VulnerabilityID
 	})
-	return vulnerabilities
+	return vulnerabilities, nil
+}
+
+func applyPolicy(ctx context.Context, vulns []types.DetectedVulnerability, policyFile string) ([]types.DetectedVulnerability, error) {
+	policy, err := ioutil.ReadFile(policyFile)
+	if err != nil {
+		return nil, xerrors.Errorf("unable to read the policy file: %w", err)
+	}
+
+	query, err := rego.New(
+		rego.Query("data.trivy.allow"),
+		rego.Module("lib.rego", module),
+		rego.Module("trivy.rego", string(policy)),
+	).PrepareForEval(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("unable to prepare for eval: %w", err)
+	}
+
+	var filtered []types.DetectedVulnerability
+	for _, vuln := range vulns {
+		results, err := query.Eval(ctx, rego.EvalInput(vuln))
+		if err != nil {
+			return nil, xerrors.Errorf("unable to evaluate the policy: %w", err)
+		} else if len(results) == 0 {
+			// Handle undefined result.
+			filtered = append(filtered, vuln)
+			continue
+		}
+		allow, ok := results[0].Expressions[0].Value.(bool)
+		if !ok {
+			// Handle unexpected result type.
+			return nil, xerrors.New("the policy must return boolean")
+		}
+		if allow {
```

Comment:
I must be missing something obvious, but why are you ignoring the vulnerability if allow==true? isn't it the opposite?

Reply:
`allow==true` means a user thinks the vulnerability is not critical and can be ignored. Is it unclear? Do you have any other good words like `accept==true`?

Reply:
Ah, you think `allow` means a user wants to display the vulnerability, right? Basically, vulnerability is not a good thing, so I think `deny=true` means dangerous and the vulnerability should be displayed.

Reply:
ah yes, since this flag was originally called "filter", I thought about the rule like "allow this vulnerability through". ignore is clear to me.
I was going to suggest this after the PR is merged but maybe it's relevant now. many times in OPA policy decisions they allow both allow and deny, so that the user can decide which strategy to use. In this case it might be something like include and exclude. I don't think it's important for the first PR, but I thought I should mention.

Reply:
If we think about the future, is `exclude==true/false` better rather than `ignore`? I have no idea how we allow both strategies, but it would be great if we can do it.

Reply:
I think I would choose "exclude" over "ignore", but it's maybe worth asking for another opinion.

If you want to allow both strategies you can add both rules to the query and then:
1. if exclude == true -> continue
2. if include == true -> append

this will implicitly handle all the other cases: exclude && !include -> continue, !exclude && include -> append, !exclude && !include -> continue.

Reply:
actually, what I suggested is not so simple, as it requires special handling of undefined by the user adding `else = false` or something similar. regardless of the technical feasibility, combining allowlist and denylist in the same policy will not be intuitive for the user, so perhaps it's better to split it into separate, mutually exclusive flags: `--include-policy/--exclude-policy`